proposals/0123-block-revenue-distribution.md

```diff
@@ -137,46 +137,64 @@ vote state field `pending_delegator_rewards` and added to the balance of vote
 account. If either of these conditions is violated, the delegator rewards amount
 MUST be burned.
 
-### Runtime: Delegator Rewards Distribution
+### Runtime: Block Revenue Distribution
 
-When calculating stake delegation rewards for a particular completed reward
-epoch, construct a list of all vote accounts that were initialized at the
-beginning of the reward epoch and had a non-zero active stake delegation. For
-each vote account, retrieve its state at the end of the reward epoch and check
-the `pending_delegator_rewards` field in its vote state. Let this value be `P`.
-If `P` is non-zero, use it to calculate rewards for each of the stake accounts
-delegated to the vote account as follows: 
+At the beginning of each epoch, pending delegator rewards will be collected from
+actively staked vote accounts into the epoch rewards sysvar account and then
+distributed to stake accounts utilizing the existing partitioned rewards
+distribution mechanism described in [SIMD-0118].
 
-1. Sum all active stake delegated to the vote account during the reward epoch
-epoch. Let this total be `A`.
+#### Delegator Rewards Calculation
 
-2. For each individual stake account, multiply its active stake from the
-reward epoch by `P`, and divide the result by `A` using integer division.
-Discard any fractional lamports.
+Delegator rewards MUST be calculated during both the beginning of the first
+block of an epoch `N` and after restarting during partitioned rewards
+distribution.
 
-After calculating all individual stake rewards, sum them to obtain `D`, the
-total distribution amount. Because of integer division, the full amount `P` may
-not be distributed so compute the amount to be burned, `B`, as the difference
-between `P` and `D`.
+For each vote account, get its total active stake delegation
+during the reward epoch `N - 1`. Let this value be `A`. If `A` is zero, skip
+this vote account (any pending delegator rewards will be distributed in a future
+epoch when the vote account has active stake.)
```

Comment:
what happens to pending delegator rewards in a vote account that has been intentionally destaked with the intention of shutting down the validator? so long as leader schedule lags stake, there's potential for delegator reward lamports to end up in a vote account with no stake

Reply:
Good point. In the future, I would like to fix the issue where leader schedule lags stake which would eliminate this issue. For now, any pending rewards for unstaked vote accounts should be directly credited to the vote account.